app/assets/javascripts/discourse/app/components/glimmer-header/topic/participant.gjs

```diff
@@ -0,0 +1,64 @@
+import Component from "@glimmer/component";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { htmlSafe } from "@ember/template";
+import icon from "discourse-common/helpers/d-icon";
+import { avatarImg } from "discourse-common/lib/avatar-utils";
+import getURL from "discourse-common/lib/get-url";
+import eq from "truth-helpers/helpers/eq";
+
+export default class Participant extends Component {
+  @service appEvents;
+
+  get url() {
+    return this.args.type === "user"
+      ? this.args.user.path
+      : getURL(`/g/${this.args.username}`);
+  }
+
+  get typeClass() {
+    return `trigger-${this.args.type}-card`;
+  }
+
+  get avatarImage() {
+    return htmlSafe(
+      avatarImg({
+        avatarTemplate: this.args.user.avatar_template,
+        size: "tiny",
+        title: this.args.username,
+      })
+    );
+  }
+
+  @action
+  click(e) {
+    this.appEvents.trigger(
+      `topic-header:trigger-${this.args.type}-card`,
+      this.args.username,
+      e.target
+    );
+    e.preventDefault();
+  }
+
+  <template>
+    <span class={{this.typeClass}}>
+      <a
+        class="icon"
+        {{on "click" this.click}}
+        href={{this.url}}
+        data-auto-route="true"
+        title={{@username}}
+      >
+        {{#if (eq @type "user")}}
+          {{this.avatarImage}}
```

Comment:
Could we use the `{{avatar }}` helper instead of manually fetching the HTML?

Reply:
This was because it was not possible to pass options to `{{avatar` and we needed to pass a `title`. I updated the helper https://github.com/discourse/discourse/pull/25214/commits/3785e36b3fc66cc46a91de7c8bebaf9fa341c104